### PR TITLE
Create api for getting gym data

### PR DESCRIPTION
**Pull request by rawaneyad**
-  Create route /api/v1/gyms/:gymID
-  Create controller
-  Handle data Form the database

and the response like this
```
{
  "GymData": {
    "id": 1,
    "gymName": "نادي فريندز للياقة البدنية",
    "logo": "https://bit.ly/3yij5Wb",
    "city": "غزة",
    "description": ".نادي رياضي يحتوي على ملعب وميدان تنافسي وملعب رياضي, أوقات الدوام من 8-11 صباحا ومن 13-17 مساء",
    "features": [
      "ميدان تنافسي",
      "ملعب رياضي"
    ],
    "images": [
      {
        "pathUrl": "https://bit.ly/39s08Wv"
      },
      {
        "pathUrl": "https://bit.ly/3kpe2e9"
      },
      {
        "pathUrl": "https://bit.ly/3LxTj3Y"
      }
    ],
    "reviews": [
      {
        "rate": 2,
        "description": "للأسف كانت تجربة سيئ جدأ , لا أنصح به ",
        "createdAt": "2022-05-08T14:15:25.265Z",
        "userId": 3,
        "user": {
          "username": "حسن عبدالله",
          "avatar": "https://bit.ly/38CU9xq"
        }
      },
      {
        "rate": 1,
        "description": "للأسف كانت تجربة سيئ جدأ , لا أنصح به ",
        "createdAt": "2022-05-08T14:15:25.265Z",
        "userId": 1,
        "user": {
          "username": "محمود علي",
          "avatar": "https://bit.ly/37THiXV"
        }
      },
      {
        "rate": 3,
        "description": "للأسف كانت تجربة سيئ جدأ , لا أنصح به ",
        "createdAt": "2022-05-08T14:15:25.266Z",
        "userId": 2,
        "user": {
          "username": "علي محمود",
          "avatar": "https://bit.ly/3EYZU4G"
        }
      }
    ]
  }
}
```

### CODE REVIEW

**Review by AhmedQeshta — APPROVED** [review]

File: server/controllers/gym/getGym.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
+import { NextFunction, Response, Request } from 'express';
+import { Gym, Image, Review, User } from '../../database/models';
+
+export default async function getGym(req: Request, res: Response, next: NextFunction) {
+  try {
+    const { id } = req.params;
```
Comment (LinaYahya) [inline server/controllers/gym/getGym.ts:6]: validate params to be a number using `hapi/joi` before running database find query

File: server/controllers/gym/getGym.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    const Gymdata = await Gym.findAll({
```
Comment (LinaYahya) [inline server/controllers/gym/getGym.ts:7]:
```suggestion
    const gymData = await Gym.findAll({
```
Follow camelCase convention naming your vars

File: server/controllers/gym/getGym.ts [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+      subQuery: false,
+      attributes: [
+        'id',
+        ['gym_name', 'gymName'],
```
Comment (LinaYahya) [inline server/controllers/gym/getGym.ts:11]: change it in the Gym model itself

File: server/routes/gyms.ts [hunk 4]
```diff
@@ -1,8 +1,8 @@
 import { Router } from 'express';
-import getTopGyms from '../controllers/index';
+import { getAllGyms, getGym } from '../controllers/index';
 
 const gyms = Router();
 
-gyms.get('/top', getTopGyms);
-
+gyms.get('/top', getAllGyms);
```
Comment (LinaYahya) [inline server/routes/gyms.ts:6]: `getTopGyms` more descriptive I think

File: test/routes.test.ts [hunk 5]
```diff
@@ -10,6 +10,26 @@ describe('Gyms API', () => {
     const response = await request(app).get('/api/v1/gyms/top').expect(200);
     expect(response.body.topReviewGyms.length).toBe(3);
   });
+  test('Gyms - GET - /api/v1/gyms/:id - ', async () => {
+    const response = await request(app).get('/api/v1/gyms/1').expect(200);
+    expect(response.body.Gymdata.length).toBe(1);
+  });
+  test('Gyms - GET - /api/v1/gyms/:id - ', async () => {
```
Comment (LinaYahya) [inline test/routes.test.ts:17]: add  tests to test Both `/api/v1/gyms/gggggggg` and `/api/v1/gyms/50`

**Review by LinaYahya — CHANGES_REQUESTED** [review]

**Review by AFahgan — APPROVED** [review]
Great Job 
Conflict resolved.

**Review by LinaYahya — APPROVED** [review]

**Pull request merged**